src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -338,6 +338,7 @@ internal sealed class MemberSignatureComparer : IEqualityComparer<Symbol>
             considerTypeConstraints: false,
             refKindCompareMode: RefKindCompareMode.ConsiderDifferences,
             considerCallingConvention: false,
+            considerArity: false,
```

Comment:
:memo: although the legacy logic uses this comparer, it is not affected by this change since it only performs substitution/reduction once a unique signature is found.